File: python/qos.py
#!/usr/bin/env python
# Check that QoS information is displayed
import subprocess as sp
p = sp.Popen(['sacctmgr', 'show', 'qos'], stdout=sp.PIPE, stderr=sp.PIPE)
# @todo check stderr
if len(str(p.communicate()[0]).split('\\n')) > 2:
    print('PASS')
else:
    print('FAIL')

File: rust/slurm-test/rust/slurm-test/src/bin/qos.rs
use subprocess::{Popen, PopenConfig, Redirection, Result};
/// Verify that quality of service is supported by checking
/// that the output contains more than the two header rows.
fn main() -> Result<()> {
    let mut p = Popen::create(
        &["sacctmgr", "show", "qos"],
        PopenConfig {
            stdout: Redirection::Pipe,
            stderr: Redirection::Pipe,
            ..Default::default()
        },
    )?;
    let (out, err) = p.communicate(None)?;
    if let Some(e) = err {
        if !e.is_empty() {
            eprintln!("ERROR: {}", e);
            std::process::exit(1);
        }
    }
    if let Some(c) = out {
        let numrows = c
            .split('\n')
            .skip(2)
            .filter(|line| !line.is_empty())
            .count();
        if numrows > 0 {
            println!("PASS");
        } else {
            println!("FAIL");
        }
    } else {
        eprintln!("ERROR");
    }
    Ok(())
}
